### PR TITLE
CI/CD workflow to publish helm chart

### PR DESCRIPTION
**Pull request by robinson**
**Many thanks for submitting your Pull Request :heart:!**

**What this PR does / why we need it**: CI/CD flow to publish helm chart

**Special notes for reviewers**: need another step to publish into https://artifacthub.io/

**Additional information (if needed):**

### CODE REVIEW

**Comment by JBBianchi** [comment]
Hi @robinson, thanks a lot for your PR !

But as you can see, it doesn't pass the DCO requirements. Could you please sign/verify and sign off your commits ?

**Review by cdavernas — APPROVED** [review]
LGTM! Cheers ❤️

File: deployments/helm/Chart.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 apiVersion: v2
 name: synapse
 description: A Helm chart for deploying Synapse
-version: 0.1.0
-appVersion: "1.0.0"
\ No newline at end of file
+version: 1.0.0-alpha5.1
```
Comment (cdavernas) [inline deployments/helm/Chart.yaml:4]: Will we have to update this manually upon every release? If so, isn't there a way to automatize it using the CI pipeline?

**Comment by JBBianchi** [comment]
Btw, should this PR be merged in main or in [gh-pages](https://github.com/serverlessworkflow/synapse/tree/gh-pages) ?

File: deployments/helm/Chart.yaml (hunk 1 above)
Comment (robinson) [reply deployments/helm/Chart.yaml:4]: let me making a test it with auto version base on the release.

**Comment by robinson** [comment]
> Btw, should this PR be merged in main or in [gh-pages](https://github.com/serverlessworkflow/synapse/tree/gh-pages) ?

the gh-pages branch use to create a pages in order to publish the artifacts. The helm release workflow should be run in main branch.

File: deployments/helm/Chart.yaml (hunk 1 above)
Comment (robinson) [reply deployments/helm/Chart.yaml]: looks like not trivial :-) the most popular helm release strategey should be 1-1 versioning, that mean, when new release taking place, simply create a new version helm chart folder (refer: https://codefresh.io/docs/docs/ci-cd-guides/helm-best-practices/#simple-1-1-versioning) what do you think?

Comment (cdavernas) [reply deployments/helm/Chart.yaml]: That would be great! Can you add that strategy to the pipeline?

Comment (robinson) [reply deployments/helm/Chart.yaml]: maybe I wrote not really clearly, when releasing then a new folder with same version will be created manually [helm](https://github.com/serverlessworkflow/synapse/tree/main/deployments/helm), then this pipeline will create helm package accordingly.

Comment (cdavernas) [reply deployments/helm/Chart.yaml]: Ah okok. I'm sure we can create that folder automatically, though.

**Pull request merged**